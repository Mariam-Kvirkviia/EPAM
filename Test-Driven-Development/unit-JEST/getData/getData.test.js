const axios = require("axios");
jest.mock("axios"); //mocking axios
const getData = require("./getData");
describe("get Data", () => {
  test("gd", async () => {
    let data = {
      userId: 1,
      id: 1,
      title: "delectus aut autem",
      completed: false,
    };
    await axios.get.mockReturnValue(data);

    let gd = await getData();
    expect(axios.get).toBeCalledTimes(1);
    console.log(gd);
    expect(gd).toBe(data);
    expect(gd).toMatchSnapshot();
  });
});
