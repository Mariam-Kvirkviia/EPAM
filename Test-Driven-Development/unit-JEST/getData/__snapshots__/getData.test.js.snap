// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get Data gd 1`] = `
{
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;
